src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/TransportCreateSnapshotAction.java

```diff
@@ -65,7 +65,7 @@ protected CreateSnapshotResponse newResponse() {
 
     @Override
     protected ClusterBlockException checkBlock(CreateSnapshotRequest request, ClusterState state) {
-        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA, "");
+        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA_WRITE, "");
```

Comment:
maybe we should only check for metadata_read here? then #8102 would get fixed too. create snapshot only writes to the repo right? it's not really a write operation within the cluster? As long as you can read the data and metadata, you should be able to snapshot no? which makes me wonder, do we check for index read blocks on data?

@imotov thoughts?